### PR TITLE
Update device authorization and JWKS to closer match the spec

### PR DESCRIPTION
**Pull request by mikenachbaur-okta**
This matches up to https://github.com/okta/okta-mobile-kotlin/pull/183 and https://github.com/okta/okta-mobile-kotlin/pull/182

### CODE REVIEW

**Review by JayNewstrom — COMMENTED** [review]
I think the one missing thing from those 2 PRs is default scope.

https://github.com/okta/okta-mobile-swift/blob/master/Sources/AuthFoundation/Token%20Management/Token.swift#L168

I choose null, but there might be a more correct answer.

The rest looks good though.

**Comment by mikenachbaur-okta** [comment]
Yeah, that seems like the right thing to do. Also, Okta has a feature for defining a default scope for when a client doesn't supply one, so we might want to make the `scope` argument optional for our authorization flows.

File: Tests/AuthFoundationTests/TokenTests.swift [hunk 1]
```diff
@@ -47,6 +48,22 @@ final class TokenTests: XCTestCase {
         XCTAssertEqual(context, decodedContext)
     }
     
+    func testNilScope() throws {
+        let data = data(for: """
+        {
+           "token_type": "Bearer",
+           "expires_in": 3600,
```
Comment (JayNewstrom) [inline Tests/AuthFoundationTests/TokenTests.swift:55]: Worth noting, some Authorization servers may not return expires_in (notably when you are doing pure OAuth, without OIDC). I don't think it's something we need to account for, but I think the only 2 things that are required from OAuth are access_token/token_type.

Comment (mikenachbaur-okta) [reply Tests/AuthFoundationTests/TokenTests.swift:55]: Yeah, I'm not sure if we should worry about that. There's a lot of code that assumes the `expires_in` exists, for both Swift and Kotlin SDKs.  So I'd hold off on this, until it's necessary.

**Pull request merged**